src/filesystem/implementations/s3.h

```diff
@@ -39,6 +45,50 @@ namespace triton { namespace core {
 
 namespace s3 = Aws::S3;
 
+// Override the default S3 Curl initialization for disabling HTTP/2 on s3.
+// Remove once s3 fully supports HTTP/2 [FIXME: DLIS-4973].
+// Reference:
+// https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/cpp/example_code/s3/list_buckets_disabling_dns_cache.cpp
+static const char S3_ALLOCATION_TAG[] = "OverrideDefaultHttpClient";
+class S3CurlHttpClient : public Aws::Http::CurlHttpClient {
+ public:
+  explicit S3CurlHttpClient(
+      const Aws::Client::ClientConfiguration& client_config)
+      : Aws::Http::CurlHttpClient(client_config)
+  {
+  }
+
+ protected:
+  void OverrideOptionsOnConnectionHandle(CURL* connectionHandle) const override
+  {
+    curl_easy_setopt(connectionHandle, CURLOPT_HTTP_VERSION, 2L);  // HTTP 1.1
```

Comment:
Do we want to force HTTP 2 or HTTP 1.1? Does 2L resolve to CURL_HTTP_VERSION_1_1? If so, maybe update it to `CURL_HTTP_VERSION_1_1` to avoid confusion.

Reply:
@kthui if the constant isn't available - can we define it ourselves at the top - agree with @Tabrizian for readability - having a named constant here is beneficial

Reply:
> Do we want to force HTTP 2 or HTTP 1.1?

We want to limit the version to http/1.1 (or http/1.0), but not http/2.0 or above.

> Does 2L resolve to CURL_HTTP_VERSION_1_1?

Yes.

> If so, maybe update it to `CURL_HTTP_VERSION_1_1` to avoid confusion.

Yes, the build is successful with `CURL_HTTP_VERSION_1_1` constant and passing the test locally.